### ISSUE TITLE
Request.path includes query string

### ISSUE THREAD

**Issue by 1268** [mitmproxy/mitmproxy#6345]
using this script:

~~~py
from mitmproxy import http
import logging

def request(f: http.HTTPFlow) -> None:
   logging.info(f.request.path)
~~~

I am getting results like this:

~~~
/fdfe/acquire?theme=2
~~~

this is surprising to me, because the path is also including the query string. this runs counter to the definition in RFC 3986:

https://rfc-editor.org/rfc/rfc3986.html#section-3

as well as Pythons own `urllib.parse`:

https://docs.python.org/library/urllib.parse.html#urllib.parse.urlparse

I think the `path` should be returned without a query string, or at least it should be documented that a `path` might include a query string:

https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Request.path

**Comment by mhils** [comment]
You're not wrong here, but for backward compatibility reasons I would like to retain the current behavior. I'd be very happy to merge a PR that updates the documentation though! 😃

**Comment by 3052** [comment]
am I missing something? the docs seem to be the same, so what was changed?

**Comment by mhils** [comment]
https://docs.mitmproxy.org/dev/api/mitmproxy/http.html#Request.path is updated, /stable/ will follow with the next release. :)
